src/daemon/daemon.cpp

```diff
@@ -124,7 +124,7 @@ bool t_daemon::run(bool interactive)
       return false;
     mp_internals->rpc.run();
 
-    daemonize::t_command_server* rpc_commands;
+    daemonize::t_command_server* rpc_commands = NULL;
```

Comment:
I would encourage everyone to become C++ programmers in this situation. Monero technically has lots of potential memory leaks because of exceptions ...

If any function between the `new` and `delete` throws, the memory still never gets cleaned up. You either have to: (1) add `try{} catch(...) {}` blocks around every function call, (2) inspect every code path for every function and only add tries around functions that can throw, or (3) use C++ smart pointers and force the compiler to emit the necessary cleanup code. The last approach is the easiest to get correct (and IMO, when done correctly yields in less memory leaks than C when maintaining large code bases), BUT now the compiler pessimistically emits memory cleanup code for every non-inlined function that does not have a `noexcept` or `throw()` specifier. So if those specifiers aren't aggressively used, potential code generation bloat. Great stuff.

Anyway in this circumstance its not huge because the process should be shutting down, but its worth understanding unless exceptions are disabled.